### ISSUE TITLE
Cannot sign into Onshape

### ISSUE THREAD

**Issue by Buyingbf** [wawanbreton/CuraOnshapePlugin#1]
Hi, when I try to sign into Onshape through the plugin, I get an error 403 with the message "Resource does not exist, or you do not have permission to access it." I am signed into both Onshape and Cura 5.8.1. I see the Onshape Cura plugin in my Onshape applications, but the Onshape access and External access fields are both blank.

**Comment by wawanbreton** [comment]
Hi @Buyingbf, sorry I didn't answer sooner, I didn't get a notification that you opened the issue.

Normally, the first time that you try to access OnShape via Cura, the browser should display a page to allow the plugin to access your OnShape account. But if you see it on your applications, that means you have successfully been through that process. For me, the two fields are also blank, so I assume this is normal.

However, I am currently in contact with the Onshape support because they told me the plugin was spamming their auth server. Maybe this is related to the issue you have. If you have a 403 error, maybe the plugin tries to authenticate repeatedly. I'll check that everything is properly setup on my side for other people to use it.

**Comment by wawanbreton** [comment]
Hi, I am not making much progress with the Onshape support. Could you open a ticket on their assisitance and try to figure out why you get the 403 error ? I have a report of an other user for whom it is working as expected :thinking:

**Comment by Buyingbf** [comment]
Hi, the issue is still present for me even after not using Onshape and Cura for more than a month, so I'll put a ticket in and see what's happening.

**Comment by Buyingbf** [comment]
The Onshape support rep I am talking to suggests that there is some kind of authentication issue with me and plugin, and not with them. In my error message, the rep mentioned that the requested API folder is owned by you and suggests sharing it with me. 
![err](https://github.com/user-attachments/assets/0ac2969d-5207-4193-9f5a-9a862b765840)

**Comment by wawanbreton** [comment]
Ok, I don't know what could be wrong on my side, because we are now a few users for who the authentication works, so no idea why it doesn't for you. I'll open a ticket on the Onshape support to try and figure it out. Could you eventually send me your onshape username, if it is not private ?

**Comment by Buyingbf** [comment]
My forum username is rafael_baltazar

**Comment by wawanbreton** [comment]
Hi, I have contacted the Onshape support but it is a bit vague right now. Can you please:
* Send me the cura logs (or a project exported with "Help" > "Export Package for Technical Support" if you are using 5.9 Beta)
* Record a video of your steps before you get the error

**Comment by wawanbreton** [comment]
Also, can you tell me if you have a regular account, or maybe sone kind of company account with limited rights ?

**Comment by Buyingbf** [comment]
I've uploaded my video and log to my [google drive](https://drive.google.com/drive/folders/1eOdx4mMxANQQua6ymuG_JGrduVP0kKx4?usp=sharing). My account should also be a free personal account since I made it with my personal email account.

**Comment by wawanbreton** [comment]
Thanks ! That eliminates some possible issues, but I still don't see what is going wrong.

The Onshape support asks whether you have properly authorized the application, but from what we could see, this is good. I asked them to check your account, but not sure they can do that.

The only advice I can give you right now is to clear the authentication data in the cura configuration:
* Find the `cura.cfg` file, which should be somewhere in C:\Users\...\AppData\Roaming\cura\5.9
* Remove the whole section "plugin_onshape" and all its content, then save the file
* Restart Cura. It should ask for a brand new Onshape authentication, hopefully that could help :crossed_fingers:

**Comment by Buyingbf** [comment]
I was able to do what you suggested but I still have the issue.

**Comment by wawanbreton** [comment]
Well, the support keeps telling me that you need to grant the application to allow it to use the Onshape API. There should be a button somewhere in your Onshape settings, named "Grant". But I do trust that you did that already. Could you send a screenshot of your Onshape application page ?

**Comment by Buyingbf** [comment]
![image](https://github.com/user-attachments/assets/49daf24e-87cf-471b-8959-a56dcb17826a)
This is what I see in my Applications page, no Grant button is present.

**Comment by wawanbreton** [comment]
Yeah it does look very much like what I have. I'll transfer this to the support and see what they say next.

**Comment by wawanbreton** [comment]
Ok, new answer from the support :slightly_smiling_face: 
Is it possible that you have a document in your list that you don't own ? It could be a document owned by someone else that you have access to, or a document that links to an other one, owned by someone else.

**Comment by Buyingbf** [comment]
That seems to have worked! First, I removed some files that I owned but were labelled as "created by an education subscriber" (I guess I was subscribed at some point?) but that didn't work. I then found a couple files shared to me from about 3 years ago (also created by an education subscriber, not sure if that's relevant) and I removed my access from those files. Now, the plugin is working as intended.

**Comment by wawanbreton** [comment]
Oh, I'm very glad to hear that :slightly_smiling_face: 
That means I will have to make a fix then, in your case it may not be an issue to remove the linked documents, but in others it may be. I should be able to reproduce it easily though !

**Comment by dgosgd** [comment]
Hello hello! I am having the same issue that Buyingbf was, same exact error and everything. Now I *am* an education subscriber (HS Tech teacher). Is it the case at this point that the plugin just does not work if you are an education subscriber? I also have Onshape docs that are def shared with me, but I don't want to unshare them. Is it the case that the plugin just won't work for my account then? 

Thanks!

**Comment by wawanbreton** [comment]
Hi @dgosgd, from what I understand, education accounts should work, but if you have documents that are shared from someone else, that won't until I fix the plugin.

**Comment by dgosgd** [comment]
Ah ha! Okay, thanks! I do think that I probably have shared docs that at
this time I don't want to unshare, but that is okay, thank you!

Doug Osgood (he/him/his)
Tech and Engineering Ed Teacher
C-PP HS
Corning, NY

**Comment by wawanbreton** [comment]
Hi, I have tried to reproduce this issue by asking someone to share a document with me. But in my case, it all worked perfectly and I can even download parts from the shared document. Could someone please re-test if in case there has been some API change that would have fixed it ?

**Comment by Buyingbf** [comment]
Hi, I have been using the plugin without issue for months, even importing a file that has been shared to me. However, I just realized today that the aforementioned shared file was also created by an education subscriber. So I am completely unsure what my problem was before.

**Comment by wawanbreton** [comment]
Ok, thanks for the feedback. So that may have to see with the education accounts as suggested. @dgosgd as you have such an account, could you please share any document you have with me ? My email is my github username - at - gmail.com
